crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1050,6 +1096,40 @@ impl AuthorityStore {
             .cloned()
             .collect();
 
+        let mut output = Vec::new();
+        let _ = deleted
+            .iter()
+            .filter(|(object_id, _)| {
+                let object = objects.get(object_id).unwrap();
+                object.is_shared()
+            })
+            .map(|(object_id, (version, _))| {
+                output.push((object_id, version));
+            });
+
+        // get shared deleted objects and write them to the object marker table
+        write_batch.insert_batch(
```

Comment:
When a transaction uses an already deleted shared object, will it show up again in deleted in the effects?

Reply:
no it will not, in the case the deletion was already executed, this part will filter out that object 
```
deleted.filter(|(object_id, _)| {
                    if let Some(object) = objects.get(object_id) {
                        return object.is_shared();
                    }
                    false
                })